src/fundus/parser/data.py

```diff
@@ -113,7 +115,15 @@ def to_unicode_characters(text: str) -> str:
             self.__xml = lxml.etree.fromstring(xml)
         return self.__xml
 
-    def xpath_search(self, query: XPath) -> List[Any]:
+    @overload
+    def xpath_search(self, query: Union[XPath, str], scalar: Literal[False] = False) -> List[Any]:
+        ...
+
+    @overload
+    def xpath_search(self, query: Union[XPath, str], scalar: Literal[True] = True) -> Optional[Any]:
+        ...
+
+    def xpath_search(self, query: Union[XPath, str], scalar: bool = False) -> Union[Any, List[Any]]:
```

Comment:
Not being that familiar with typing, is this third variation necessary for the case where a variable value is passed to scalar? So it's undetermined at compilation, whether it will be True or False?

Also, if this is the case, shouldn't the return value also be `Union[Optional[Any], List[Any]]`, to be consistent with the case of Literal[True]?

Reply:
No, you don't have to type hint the original function, it's just for IDEs like Pycharm.

You're right, I forgot the `Optional` there.